File: public/src/types/Product.ts
export const CATEGORY_LABELS = {
  CANDY: "Doce",
  SAVORY: "Salgado",
  CAKE: "Bolo",
} as const;

export type CategorySlug = keyof typeof CATEGORY_LABELS;

export interface Product {
  id: string;
  name: string;
  description: string;
  price: number;
  image: string;
  category: CategorySlug;
  featured?: boolean;
  ingredients: string[];
}

File: app/src/main/java/com/example/demo/models/Product.java
package com.example.demo.models;

import com.example.demo.enums.ProductCategory;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.Set;
import java.util.UUID;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "TB_Product")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "name", nullable = false, unique = true)
    private String name;

    @Column(name = "description", nullable = false, unique = true)
    private String description;

    @Column(name = "price", nullable = false)
    private BigDecimal price;

    @Column(name = "image", nullable = false, columnDefinition = "TEXT")
    private String image;

    @Column(name = "category", nullable = false)
    @Enumerated(EnumType.STRING)
    private ProductCategory category;

    @Column(name = "featured", nullable = false)
    private Boolean featured;

    @Column(name = "ingredients", nullable = false)
    private Set<String> ingredients;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive;

}